ui/src/dropdown.tsx

```diff
@@ -152,119 +169,160 @@ const
   },
   ROW_HEIGHT = 44,
   PAGE_SIZE = 40,
-  DialogDropdown = ({ name, choices, values, value, disabled, required, trigger, placeholder, label }: Dropdown) => {
+  getPageSpecification = () => ({ itemCount: PAGE_SIZE, height: ROW_HEIGHT * PAGE_SIZE } as Fluent.IPageSpecification),
+  useItems = (choices: Choice[], v?: S | S[]) => {
     const
-      isMultivalued = !!values,
-      [isDialogHidden, setIsDialogHidden] = React.useState(true),
-      initialSelectedMap = React.useMemo(() => {
-        if (values?.length) return new Map(values.map(v => [v, true]))
-        if (value) return new Map([[value, true]])
-        return new Map()
-      }, [value, values]),
-      items = React.useMemo<DropdownItem[]>(() => choices?.map(({ name, label }, idx) => ({ name, text: label || name, idx, checked: initialSelectedMap.has(name) })) || [], [initialSelectedMap, choices]),
-      [filteredItems, setFilteredItems] = React.useState(items),
-      [textValue, setTextValue] = React.useState(() => {
-        if (!values?.length && !value) return
-
-        const itemsMap = new Map<S, S>(items.map(({ name, text }) => [name, text]))
+      [items, setItems] = React.useState<DropdownItem[]>(choices.map(({ name, label }, idx) =>
+        ({ name, text: label || name, idx, checked: Array.isArray(v) ? v.includes(name) : v === name, show: true }))),
+      onSearchChange = (_e?: React.ChangeEvent<HTMLInputElement>, newVal = '') => setItems(items => items.map(i => ({ ...i, show: fuzzysearch(i.text, newVal) })))
+    
+    return [items, setItems, onSearchChange] as const
+  },
+  onRenderCell = (onChecked: any) => (item?: DropdownItem) => item
+    ? <Fluent.Checkbox
+      label={item.text}
+      styles={{
+        root: { minHeight: ROW_HEIGHT, alignItems: 'center' },
+        label: { width: pc(100), padding: 12, wordBreak: 'break-word' },
+        checkmark: { display: 'flex' }
+      }}
+      onChange={onChecked(item.name)}
+      className={item.checked ? css.dialogCheckedRow : ''}
+      checked={item.checked} />
+      : null,
 
-        if (values?.length) return values.map(v => itemsMap.get(v) || '').filter(Boolean).join(', ')
-        if (value) return itemsMap.get(value)
-      }),
-      toggleDialog = React.useCallback(() => setIsDialogHidden(!isDialogHidden), [isDialogHidden]),
-      cancelDialog = React.useCallback(() => {
+  DialogDropdownSingle = ({ model }: { model: Dropdown}) => {
+    const
+      { name, choices = [], disabled, required, trigger, placeholder, label } = model,
```

Comment:
nit: maybe we can use destructuring assignment for `value` here in the same way as in `BaseDropdown`?

Reply:
No, I am referering to 
![image](https://user-images.githubusercontent.com/64769322/192238110-68b16c0d-a205-4e25-a3c7-96cc3993e34a.png)

if it can be the same as BaseDropdown - your most recent impl.

![Screen Shot 2022-09-26 at 11 10 04 AM](https://user-images.githubusercontent.com/64769322/192238531-27388bca-8641-4979-b4f7-d144a1a3d156.png)

Reply:
I see, thanks for clarification.

Reply:
No problem.